backend/src/test/java/codezap/template/service/SourceCodeServiceTest.java

```diff
@@ -0,0 +1,441 @@
+            Thumbnail thumbnail = thumbnailRepository.save(new Thumbnail(template, thumbnailSourceCode));
+
+            List<Long> deleteSourceCodeIds = List.of(thumbnailSourceCode.getId());
+            UpdateSourceCodeRequest updateRequest2 = getUpdateSourceCodeRequest(othersourceCode);
+            UpdateTemplateRequest updateTemplateRequest = getUpdateTemplateRequest(
+                    Collections.emptyList(),
+                    List.of(updateRequest2),
+                    deleteSourceCodeIds,
+                    template.getCategory().getId(),
+                    Collections.emptyList()
+            );
+
+            // when
+            sourceCodeService.updateSourceCodes(updateTemplateRequest, template, thumbnail);
+
+            // then
+            Thumbnail updatedThumbnail = thumbnailRepository.fetchByTemplate(template);
+            assertAll(
+                    () -> assertThat(updatedThumbnail.getSourceCode().getId()).isEqualTo(othersourceCode.getId()),
+                    () -> assertThat(updatedThumbnail.getSourceCode().getOrdinal()).isEqualTo(1)
+            );
+        }
+
+        @Test
+        @DisplayName("성공: 소스 코드 추가 & 기존 소스 코드들 순서를 추가된 소스 코드보다 나중으로 변경")
+        void updateSourceCodes_WhenChangeOrderToLast_AndAddNewCode() {
+            // given
+            Template template = createTemplate();
+            SourceCode sourceCode1 = sourceCodeRepository.save(SourceCodeFixture.get(template, 1));
+            SourceCode sourceCode2 = sourceCodeRepository.save(SourceCodeFixture.get(template, 2));
+            Thumbnail thumbnail = thumbnailRepository.save(new Thumbnail(template, sourceCode1));
+
+            UpdateSourceCodeRequest ordinalUpdateRequest = new UpdateSourceCodeRequest(sourceCode1.getId(), "변경된 제목1",
+                    "변경된 내용1", 3);
+            UpdateSourceCodeRequest updateRequest2 = getUpdateSourceCodeRequest(sourceCode2);
+            CreateSourceCodeRequest createRequest = new CreateSourceCodeRequest("새로운 제목3", "새로운 내용3",
+                    sourceCode1.getOrdinal());
+            UpdateTemplateRequest updateTemplateRequest = getUpdateTemplateRequest(
+                    List.of(createRequest),
+                    List.of(ordinalUpdateRequest, updateRequest2),
+                    Collections.emptyList(),
+                    template.getCategory().getId(),
+                    Collections.emptyList());
+
+            // when
+            sourceCodeService.updateSourceCodes(updateTemplateRequest, template, thumbnail);
+
+            // then
+            assertAll(
+                    () -> assertThat(sourceCodeRepository.countByTemplate(template)).isEqualTo(3),
+                    () -> assertThat(sourceCodeRepository.fetchById(sourceCode1.getId()).getOrdinal()).isEqualTo(3),
+                    () -> assertThat(
+                            sourceCodeRepository.fetchByTemplateAndOrdinal(template, 1).getFilename()).isEqualTo(
+                            "새로운 제목3")
```

Comment:
똑같은 줄바꿈 피드백인데 놓칠까봐 잡아놓습니다! 마찬가지로 선택은 몰리가 편한대로!!
```suggestion
                    () -> assertThat(sourceCodeRepository.fetchByTemplateAndOrdinal(template, 1).getFilename())
                    .isEqualTo("새로운 제목3")
```